### ISSUE TITLE
Mud clues - skipping over (unplayable cards) change to (known non-muddy cards)

### ISSUE THREAD

**Issue by mangopie1** [hanabi/hanabi.github.io#852]
Inspired by this chain
https://hanab.live/replay/709614#42
https://discord.com/channels/140016142600241152/936275023243673641/936377281453129788
https://hanabi.github.io/docs/variant-specific/muddy-rainbow-cocoa-rainbow/#mud-clues-skipping-over-unplayable-cards

I think we should count all cards that could be muddy, even that they are known unplayable, so we can focus them to make them playable.

I propose to change the convention to only skip known non-muddy. Such as cards already touched by rank, or got negative color clue. 
I also feel that this is already how we played, and less prone for mistake. As everyone could be de-synced on whether a card is "known unplayable" or not.

**Prompt+Finesse Example:**
Bob: m4-m1-x-x-x
Cathy: x-x-(r)m2-(r)m3-(r)m5
m5 is known as 5 from the previous 5ce. m1 is on stack.
Alice clue purple (rightmost color) should focus m5, and Bob should assume he got m4 on finesse.
Alice avoided bad touching m1 :)
In the old convention, Alice only option is to give red to Bob, prompting m2,m3, causing a bad touch on m1.

**Double Bluff Example:**
Donald: x-x-(r)m3-(r)m5
m5 is known as 5 from the previous 5ce. m1 is on stack.
Alice clue purple (rightmost color) should focus m5, and Bob and Cathy played as double bluff.
In the old convention, focus is only m3, and will only cause Bob to blind play.
In the new convention, Alice have a free choice in focusing either m3, or m5, for single or double bluff.

**Comment by EhDrone** [comment]
I think that the confusing edge case of rainbow 5s and muddy rainbow shouldn't dictate how the convention is applied and useful for muddy cards.

The benefit of having mud cards skip over known not playables is that it increases the wrap-around in almost every case of colors that can be clued to get certain slots. Playing muddy cards is a burden and valuable even just 1 at a time - so squeezing efficiency out of being able to finesse or prompt with a known muddy 5 seems less valuable to playing muddy cards in most instances than being able to skip the muddy 5 and have potentially an extra color to hit the muddy card you actually want played. 

Similarly here is a different take on your examples where being able to reclue the m5 is irrelevant since its already known m5:
**Promt + Finesse Example:**
Bob: m4-m1-x-x-x
Cathy: x-x-(r)m2-(r)m3-(r)m5
I think bad touching the m1 is a minor sacrifice at the cost of a) getting an additional color to potentially clue m3 if m2 was in the other order, and 2) muddy cards are incredibly likely to fix for free as soon as _any other color clue is given to Bob_.

**Double Bluff Example:**
Donald: x-x-(r)m3-(r)m5
m5 is known as 5 from the previous 5ce. m1 is on the stack.
Alice clue a color matching Bob's card should focus m3 always, and Bob and Cathy play as a pestilent double bluff for the same effect, plus m3 is now known m3.

Of course these are limited examples and we could come up with 100 more in either way. Based on my muddy rainbow games, there have been many occasions where someone is holding too many muddy cards for me to correctly focus the card that I want. If that player KNEW one of them was the muddy 5, it would have been hugely helpful to be able to skip over it in a muddy clue. I think it comes down to the value or priority to getting muddy cards played vs squeezing a little bit of rare efficiency out of this.

**Comment by mangopie1** [comment]
I agree the efficiency gain is rare.

> If that player KNEW one of them was the muddy 5

However I think the main reason of wanting to change is for reducing mistakes.
As "known unplayable" could sometimes be by context, and therefore more prone for mistakes.

Example of how "known unplayable" could be confusing.
Cathy: (r)m3-(r)m2-(r)m4-x-x
m1 and r1 is down. m3 is known 3 from the previous bluff. 
Now - does mud clue disregard m3 or not?

Having the convention to be non-contextual is less confusing in my opinion.

**Comment by iamwhoiamhahaha** [comment]
This question has been brought up a couple of times before (in a review) and it's always because of 5ce 😅 (We all agreed that the 5 known from the 5ce should be skipped but people always forgot it, including myself)
I didn't notice that 5ce would be a problem while inventing the mud clues. I believe we need to redefine the rules or at least make a clarification for the interaction between mud clues and 5ce (and 4 charms I guess). I'm fine with both interpretation.

**Comment by Zamiell** [comment]
i agree with adrone

TLDR:
- the convention should be kept as is
- only cards that are explicitly unplayable (from clues) should be skipped over
- thus, 5's that are known from 5CE should always be considered as potentially muddy cards

**Comment by EhDrone** [comment]
You agree with me - but then your third point seems to be opposite of what I was saying. I am saying that a 5 known from 5ce should be skipped when considering which slot to play when given a muddy color positional clue. Is that an accurate statement from yours, or is there a disconnect?

**Comment by Zamiell** [comment]
>  I am saying that a 5 known from 5ce should be skipped when considering which slot to play when given a muddy color positional clue.

i disagree with this.
if the convention is to always treat them as muddy rainbow cards, then the consequence is that we have eliminated the vector for a mistake where someone forgets to skip over it, resulting in an overall +win rate

**Comment by EhDrone** [comment]
K just wanted to clarify. Then I would agree the convention language should update to reflect that. To me, a card is 'known unplayable' if it is a KNOWN 5, even if it is muddy. So it doesn't track to include it in a muddy clue when it is 'known unplayable'. Perhaps it should reflect to idea of 'known unplayable non-muddy' cards?

**Comment by mangopie1** [comment]
Example with interaction of non muddy cards

 
Bob: (g)4-(m)2-(r)4-(m)3
All 1s is down, and all Color in Bobs hand is known.
Alice clue red to Bob, touching slot 2,3,4.

Now for the purpose of mud clue, does it include r2?

Old convention:
r2 is not known playable, hence should be counted.

Proposed convention:
r2 is known not muddy. Hence should not be counted.

I would argue that proposed solution is clearer

**Comment by Zamiell** [comment]
then how are you supposed to tempo clue true red cards

**Comment by Zamiell** [comment]
do we have any consensus on what to do

**Comment by iamwhoiamhahaha** [comment]
Bob: r5-(g)2-(m)2-(y)2
All 1s is down, except the g1
Available color clues: Red, Yellow, Green, and Blue

If Alice wants to tempo the y2 before saving the r5 (with a 5 clue), and she does a mud clue:

### Proposed Convention
`Blue: tempo the m2`
`Green: finesse the g2`
`Yellow: tempo the m2`
`Red: 5ce`
`2: finesse the g2`

=> **THERE'S NO WAY TO TEMPO THE YELLOW 2**

### Old Convention
`Blue: tempo the m2`
`Green: finesse the g2`
`Yellow: tempo the y2`
`Red: 5ce`
`2: finesse the g2`

#### Bonus

After playing the yellow 2, Bob should realize that:
- He might have something important on *Chop* and that's the reason Alice tempos the y2
- His unknown green card is probably a 2 and that's the reason Alice is not able to tempo the y2 with a 2 clue

**Comment by EhDrone** [comment]
You need to revise the example unless I am missing something - because who cares if Alice can tempo the y2? the m2 is also playable... so just tempo that card instead, and tempo the y2 the turn after?

**Comment by iamwhoiamhahaha** [comment]
Bob: r5-(g)2-(m)3-(y)2
All 1s is down, except the g1
Available color clues: Red, Yellow, Green, and Blue

If Alice wants to tempo the y2 before saving the r5 (with a 5 clue), and she does a mud clue:

### Proposed Convention
`Blue: finesse the m3`
`Green: finesse the g2`
`Yellow: finesse the m3`
`Red: 5ce`
`2: finesse the g2`

=> **THERE'S NO WAY TO TEMPO THE YELLOW 2**

### Old Convention
`Blue: finesse the m3`
`Green: finesse the g2`
`Yellow: tempo the y2`
`Red: 5ce`
`2: finesse the g2`

#### Bonus

After playing the yellow 2, Bob should realize that:
- He might have something important on *Chop* and that's the reason Alice tempos the y2 (as a soft tccm)
- His unknown green card is probably a 2 and that's the reason Alice is not able to tempo the y2 with a 2 clue

**Comment by EhDrone** [comment]
Alternate example that perfectly fits the way things are currently configured / asking to be updated from the PR:

All 1's are down.

Bob: r5-b1-(m)3-(y)4 [with negative 2]-(m)2
Available color clues: Red, Yellow, Green, and Blue

Alice can clue yellow as a muddy clue focusing m2, because the y4 is skipped since it is known not multi / known not playable.

By changing the convention to match your specific (I claim to be) rare instance: you gain tempo-ing yellow 2 - and lose the ability to tempo m2 in my example because now the known unplayable y4 is not getting skipped.

**Comment by iamwhoiamhahaha** [comment]
Why is y4 not skipped in current convention? It still gets m2

**Comment by Zamiell** [comment]
@EhDrone

**Comment by EhDrone** [comment]
It is skipped in the current convention - like I said it fits the way things are configured currently. With the proposed convention, there would be no way to get the m2 because you aren't skipping the yellow 4 - and the only color to hit m2 with is blocked (blue).

Perhaps I am misunderstanding something from the proposal you typed above - but it seemed to me like if we change it to the Proposed text, then m2 in my example can never be tempo'd because y4 is in the way.

**Comment by mangopie1** [comment]
The proposed convention also skip over y4. Yellow is the 3rd Color so it wraps around to m2

**Comment by Zamiell** [comment]
sounds like most are against, correct me if i am wrong
